MDANSE/Src/MDANSE/Mathematics/Arithmetic.py

```diff
@@ -44,11 +54,59 @@ def get_weights(
     Tuple(Dict[Tuple[str], float], float)
         Dictionary of scaling factors per dataset key, and a sum of all the factors
     """
+    n_atms = sum(all_contents[el] for el in all_props)
+    weights, _ = adjust_weights(
+        selected_props, selected_contents, n_atms, dim, conc_exp
+    )
+    _, normFactor = adjust_weights(all_props, all_contents, n_atms, dim, conc_exp)
+
+    normalise = True
+    try:
+        len(normFactor)
+    except TypeError:
+        normalise = abs(normFactor) > 0.0  # if normFactor is 0, all weights are 0 too.
+    if normalise:
+        for k in list(weights.keys()):
```

Comment:
Because you aren't changing the keys (only the values), don't need to fix keys.
```suggestion
        for k in weights:
```